src/vector_topic_modeling/evaluation.py

```
@@ -206,21 +219,21 @@ def calculate_extended_metrics(
                 continue
             dist_ij = max(0.0, 1.0 - cosine_similarity(centroids[i], centroids[j]))
             if dist_ij == 0.0:
-                # To avoid division by zero, consider very close centroids as having a high penalty
-                r_ij = float("inf")
+                # Strong finite penalty for overlapping centroids.
+                r_ij = LARGE_DB_PENALTY
             else:
                 r_ij = (scatter[i] + scatter[j]) / dist_ij
             if r_ij > max_r:
                 max_r = r_ij
```

Ruff (0.15.7)

[warning] Replace `if` statement with `max_r = max(max_r, r_ij)`

Replace with `max_r = max(max_r, r_ij)`

(PLR1730)